puffin/src/merge.rs

```diff
@@ -215,16 +215,18 @@ mod tests {
 
             for i in 0..2 {
                 let ns = 1000 * i;
-                let a = stream.begin_scope(ns + 100, ScopeId::new(1), "");
+                let ptr = &(now_ns as fn() -> i64);
+
+                let (a, _) = stream.begin_scope(ptr, 200, ScopeId::new(1), "");
                 stream.end_scope(a, ns + 200);
 
-                let b = stream.begin_scope(ns + 200, ScopeId::new(2), "");
+                let (b, _) = stream.begin_scope(ptr, ns + 200, ScopeId::new(2), "");
```

Comment:
This feels messy.

If you pass in a closure to `begin_scope` instead, then you can use `|| ns + 200` here in the test code and pass in `|| self.now_ns()` in the real code. No need for the offset argument

Reply:
Wasnt sure if it was an intentional choice to use static fn pointer over closure?

Reply:
`NsSource` is a static `fn` pointer to avoid `Box` + `dyn` overhead, but here a closure will be monomorphized with no runtime overhead (but please verify with `cargo bench`)